### ISSUE TITLE
Documentation for realm import limitations using the new operator missing

### ISSUE THREAD

**Issue by mpreu** [keycloak/keycloak#12725]
### Description

The [documentation](https://www.keycloak.org/operator/realm-import#_applying_the_realm_import_cr) section about importing a realm using the new operator is missing currently existing restrictions:

- The inlined realm definition is limited in the nesting level of some resources like groups (and potentially components, looking at the code base) and will be silently converted into this restricted format

### Discussion

https://github.com/keycloak/keycloak/discussions/12706

### Motivation

Missing documentation leads to the problem, that a user provides a valid realm definition as an input, but the import CR ending in the cluster will differ from it. This leads to a failing import, because essential content did not make it to the actual import operation.

**Comment by andreaTP** [comment]
I cannot reproduce the issue e.g. with nested `subGroups` the following RealmImport CR:

```yaml
apiVersion: k8s.keycloak.org/v2alpha1
kind: KeycloakRealmImport
metadata:
  name: example-count0-kc
spec:
  keycloakCRName: example-kc
  realm:
    id: count0
    realm: count0
    groups:
    - id: 1f433252-3f96-44a2-95b4-db3ee2c4e224
      name: count0
      path: "/count0"
      attributes: {}
      realmRoles: []
      clientRoles: {}
      subGroups:
        - id: afd4225b-1982-478b-a3ec-0a29ba8e127e
          name: count1
          path: "/count1"
          attributes: {}
          realmRoles: []
          clientRoles: {}
          subGroups: []
```

fails the upload to Kubernetes with the expected error:
```
error validating data: ValidationError(KeycloakRealmImport.spec.realm.groups[0].subGroups[0]): unknown field "subGroups" in org.keycloak.k8s.v2alpha1.KeycloakRealmImport.spec.realm.groups.subGroups; if you choose to ignore these errors, turn validation off with --validate=false
```

**Comment by nuwang** [comment]
I just ran into this issue with groups nested at 3 levels:
```
    groups:
      - id: 65b54dfb-9942-4b59-8c70-625cae4ff4d5
        name: first
        path: /first
        attributes: {}
        realmRoles: []
        clientRoles: {}
        subGroups:
          - id: f90f9c3e-afd0-460d-a10c-474db4318817
            name: second
            path: /first/second
            attributes: {}
            realmRoles: []
            clientRoles: {}
            subGroups:
              - id: f681a665-6b7e-4dc9-b602-68dc7e3da001
                name: third
                path: /first/second/third
                attributes: {}
                realmRoles: []
                clientRoles: {}
                subGroups: []
```

The first two groups were successfully imported, but the third group was silently dropped. However, I had a user who was explicitly added to a group at the 3rd nested level, and therefore, the overall import failed.

This is with latest image version: 21.1.1

If the issue is about ensuring CRD validation, perhaps the first 2 to 3 levels could be validated, after which the data type can be an arbitrary dict? That way, the import would  succeed as expected. I would imagine that most users would prefer the import to work even if there's no validation, as in my case at least, it has thrown up a significant roadblock. The third level group is required for things to work, and I'm not sure how to proceed.

**Comment by andreaTP** [comment]
Since @shawkins is picking up the ball here, this limitation can be easily solved ( #12783 ) at least for those common use cases.

**Comment by shawkins** [comment]
> Since @shawkins is picking up the ball here, this limitation can be easily solved ( https://github.com/keycloak/keycloak/pull/12783 ) at least for those common use cases.

Let's evaluate a solution like https://github.com/fabric8io/kubernetes-client/pull/5234 - that should be able to handle this situatoin more declartively.
